cosima_cookbook/database.py

```diff
@@ -746,18 +756,10 @@ def index_experiment(files, session, expt, client=None):
 
     # index in parallel or serial, depending on whether we have a client
     if client is not None:
-        futures = client.map(index_file, files, experiment=expt)
+        futures = client.map(index_file, files, experiment=expt, session=session)
```

Comment:
Ah, so this fixes for #239 ?

Can we add a test for this?

Reply:
Not sure - the existing test passes, but I'd be very suspect at trying to pass a database connection trough dask - especially if you wanted to use distributed

Reply:
Yeah you're right, it doesn't work with threaded workers in a `LocalCluster`, get all the usual errors:
```
ERROR:root:Error indexing /g/data/ik11/outputs/mom6-om4-025/OM4_025.JRA_RYF/output062/19620101.ocean_month_z.nc: (sqlite3.ProgrammingError) SQLite objects created in a thread can only be used in that same thread. The object was created in thread id 22912382048064 and this is thread id 22909971748608.
```
so I guess then we shouldn't bother passing in session and just get rid of the client argument, unless there is some formulation that does work?

Reply:
The session is being passed here because creating temporary `CFVariable` objects and replacing them with unique values later is no longer working, instead the unique CFVariables are created inside index_file which needs the session.

Reply:
So we definitively can't use `client.map` on these now that `session` is being passed, so just scrap the `client` stuff altogether.

Reply:
Removed the client code, added a deprecation warning